### PR TITLE
major rewrite with cleaner API - version 3.0.0

### PR DESCRIPTION
**Pull request by geekflyer**
This is a rewrite of the library with a cleaner, more predictable API, cleaner code and docs.
It also addresses #22

### CODE REVIEW

**Comment by geekflyer** [comment]
one thing we can still discuss:  

The `redact` and `redactSync` methods now always expect a `string`. Passing anything else causes undefined behaviour and *may* cause an exception. Prior versions of the library used to simply return the original value "untouched" if it wasn't a `string`.

File: test/GcpDLPRedactor.test.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            ]
+          }
+        }
+      ]);
+
+    await expect(dlpRedactor.redactAsync(original)).resolves.toBe(expected);
+  });
+
+  it('should not treat input as regex', async function() {
+    const original = 'Just call (646) 846-1111 or (646) 846-3663 or (646) 846-3663.';
```
Comment (JaredBett) [inline test/GcpDLPRedactor.test.ts]: Looks like these are real phone numbers for a real company. We may want to replace these with fake phone numbers although they are public phone numbers from the company's public website. The same comment applies to the other 646 phone numbers in this test file.

File: foo.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { SyncRedactor } from './src/index';
```
Comment (JaredBett) [inline foo.ts]: Did you mean to check in this file? It looks like some test file.

Comment (geekflyer) [reply foo.ts]: oops, nope

File: test/GcpDLPRedactor.test.ts (hunk 1 above)
Comment (geekflyer) [reply test/GcpDLPRedactor.test.ts]: you're right - how did you catch that? :)

**Comment by geekflyer** [comment]
Any thoughts on the
> The redact and redactSync methods now always expect a string. Passing anything else causes undefined behaviour and may cause an exception. Prior versions of the library used to simply return the original value "untouched" if it wasn't a string.

?

**Comment by JaredBett** [comment]
That should be fine. I'm pretty confident that everywhere we are calling this library we were already passing in a string anyways.

**Pull request merged**